src/lib/y2security/lsm/selinux.rb

```diff
@@ -0,0 +1,445 @@
+      # @see Yast::Bootloader#kernel_param
+      def read_param(key)
+        Yast::Bootloader.kernel_param(:common, key.to_s)
+      end
+
+      # Whether the .autorelabel file should be relocated
+      #
+      # @see https://jira.suse.com/browse/SLE-17307
+      #
+      # @return [Booelan] true if root fs will mounted as read only, SELinux is not disabled,
+      #                   and running in initial stage; false otherwise
+      def relocate_autorelabel_file?
+        mode.to_sym != :disabled && Yast::Stage.initial && read_only_root_fs?
+      end
+
+      # Relocates the .autorelabel file from #{ROOT_AUTORELABEL_PATH} to #{ETC_AUTORELABEL_PATH} by
+      # removing the first and touching the latter.
+      #
+      # @see #save
+      # @see https://jira.suse.com/browse/SLE-17307
+      def relocate_autorelabel_file
+        log.info("Deleting #{ROOT_AUTORELABEL_PATH} file")
+        Yast::Execute.stdout.on_target!(RM_COMMAND, ROOT_AUTORELABEL_PATH)
+
+        log.info("Touching #{ETC_AUTORELABEL_PATH} file")
+        Yast::Execute.stdout.on_target!(TOUCH_COMMAND, ETC_AUTORELABEL_PATH)
+      end
+
+      # Whether the root file system will be mounted as read only
+      #
+      # @return [Booelan] true if "ro" is one of the root fs mount options; false otherwise
+      def read_only_root_fs?
+        staging_graph = Y2Storage::StorageManager.instance.staging
+        root_fs = staging_graph.filesystems.find(&:root?)
+
+        return false unless root_fs
+
+        root_fs.mount_options.include?("ro")
+      end
+
+      # Model that represents a SELinux mode
+      class Mode
+        extend Yast::I18n
+        include Yast::I18n
+
+        # @return [Symbol] the id representing the mode
+        attr_reader :id
+        alias_method :to_sym, :id
+
+        # @return [Hash{String=><String, :missing>}] options for setting the mode via kernel params
+        attr_reader :options
+
+        # Returns all known SELinux modes
+        #
+        # @return [Array<Mode>]
+        def self.all
+          ALL
+        end
+
+        # Returns all known keys for setting a SELinux mode via kernel command line
+        #
+        # @return [Array<String>]
+        def self.kernel_options
+          KERNEL_OPTIONS
+        end
+
+        # Finds a SELinux mode by its id
+        #
+        # @param id [Mode, String, Symbol, nil]
+        # @return [Mode, nil] found mode or nil if given mode id does not exists
+        def self.find(id)
+          ALL.find { |mode| mode.id == id&.to_sym }
+        end
+
+        # Constructor
+        #
+        # Intended to be used internally by the class
+        #
+        # @param id [String, Symbol] id of the mode
+        # @param name [String] the mode name, a string marked for translation
+        # @param disable [Boolean] whether the mode will be disabled or not
+        # @param enforcing [Boolean] if SELinux should be run enforcing or not
+        def initialize(id, name, disable, enforcing)
```

Comment:
NP: in subsequent PR, I'd like to _improve_ it to receive the help text too.

Reply:
I will omit the help by now. It will be preferable to add a specific card for it.